### PR TITLE
docs: update readme and cleanup old stuff

### LINKED ISSUE #174: Flush out README with usage examples

### PR DESCRIPTION
**Pull request by sanderpick**
Closes #174 

- Updates README with more relevant info and a flushed out _getting started_ section.
- Cleans up old / unused things:
  - Android
  - threads shell (to be replaced with the NodeJS REPL)
  - examples (they are dated and will be superseded by the README examples)

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -45,17 +45,17 @@ threatening, offensive, or harmful.
...
+Examples of representing a project or community include using an official
+project e-mail address, posting via an official social media account, or acting
+as an appointed representative at an online or offline event. Representation of
+a project may be further defined and clarified by project maintainers.
 
 ## Enforcement
 
 Instances of abusive, harassing, or otherwise unacceptable behavior may be
-reported by contacting the project team at contact@textile.io. All
+reported by contacting the project team at [INSERT EMAIL ADDRESS]. All
```
Comment (carsonfarmer) [inline CODE_OF_CONDUCT.md:58]: I think we should probably do what they say here ;)

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2018 Textile
+Copyright (c) 2020 Textile
```
Comment (carsonfarmer) [inline LICENSE:3]: Should this actually be 2018-2020?

File: api/pb/dart/LICENSE [hunk 3]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2018 Textile
+Copyright (c) 2020 Textile
```
Comment (carsonfarmer) [inline api/pb/dart/LICENSE:3]: same here?

File: db/db.go [hunk 4]
```diff
@@ -302,11 +302,29 @@ func (d *DB) GetDBInfo(opts ...Option) (info Info, err error) {
...
-	Schema         *jsonschema.Schema
-	Indexes        []Index
+	// Name is the name of the collection.
+	// Must only contain alphanumeric characters or non-consecutive hyphens, and cannot begin or end with a hyphen.
+	Name string
+	// Schema is JSON Schema used for instance validation.
+	Schema *jsonschema.Schema
+	// Indexes is a list of index configurations, which define how instances are indexed.
+	Indexes []Index
+	// An optional JavaScript (ECMAScript 5.1) function that is used to validate instances on write.
```
Comment (carsonfarmer) [inline db/db.go:312]: Very nice!

File: dist/README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # `threadsd`
 
-This is the [Threads](https://github.com/textileio/go-threads/tree/master/threadsd) daemon.
+This is the [ThreadsDB](https://github.com/textileio/go-threads/tree/master/threadsd) daemon.
```
Comment (carsonfarmer) [inline dist/README.md:3]: Are we doing threadsdb or threaddb? I can't remember what we landed on in the end?

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (carsonfarmer) [reply CODE_OF_CONDUCT.md:58]:
```suggestion
reported by contacting the project team at contact@textile.io. All
```

File: README.md [hunk 6]
```diff
@@ -7,72 +7,594 @@
...
+
+// Determine if an instance exists by ID
+exists, err := db.Has(context.Background(), threadID, "Persons", []string{aliceID})
+```
+
+#### Transactions
+
+ThreadsDB transactions come in two flavors: `WriteTransaction` and `ReadTransaction`.
+
+##### Write transactions
```
Comment (carsonfarmer) [inline README.md:379]: Could be nice to have something here about Discarding a transaction?

**Review by carsonfarmer — CHANGES_REQUESTED** [review]
Looks AMAZING @sanderpick, there's just one suggested change and a "proposed" nit from me.

File: dist/README.md (hunk 5 above)
Comment (sanderpick) [reply dist/README.md:3]: ThreadsDB with the plural was the consensus

File: LICENSE (hunk 2 above)
Comment (sanderpick) [reply LICENSE:3]: Will do

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (sanderpick) [reply CODE_OF_CONDUCT.md:58]: Ah yeah, copy pasta from the newly updated version

**Review by carsonfarmer — APPROVED** [review]
Thanks for the tiny nits @sanderpick, looks really good!

**Pull request merged**